doc/ztp/ztp_hld.md

```diff
@@ -0,0 +1,166 @@
+# Zero Touch Provisioning (ZTP)
+# High Level Design
+
+### Rev 1.0
+
+# Table of Contents
+  * [Revision](#revision)
+  * [Requirements](#requirements)
+  * [DHCP Options](#dhcp-options)
+  * [Flow](#flow)
+  * [Error Cases](#error-cases)
+  * [ZTP CLI](#ztp-cli)
+  * [Action Items](#action-items)
+  * [Phases](#phases)
+
+# Revision
+| Rev | Date     |  Author       | Change Description                |
+|:---:|:------------:|:------------------:|-----------------------------------|
+| v0.1 |   11/03/18   |    Simone Salman   | Initial version                   |
+| v0.2 |   11/14/18   |    Simone Salman   | Addressing Review comments        |
+
+# Requirements
+- Build option to have ZTP enabled
+- SNMP should be enabled during the ZTP process (achieved through leveraging updategraph)
```

Comment:
Please explain why SNMP is important to this process - it's not clear to me where it's used here.